sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ResolverEventSource.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.Diagnostics;
+using System;
+using System.Diagnostics.Tracing;
+
+namespace Azure.Iot.ModelsRepository
+{
+    [EventSource(Name = EventSourceName)]
+    internal sealed class ResolverEventSource : EventSource
+    {
+        // Set EventSource name to package name replacing . with -
+        private const string EventSourceName = "Azure-Iot-ModelsRepository";
+
+        // Event ids defined as constants to makes it easy to keep track of them
+        private const int InitFetcherEventId = 1000;
+        private const int ProcessingDtmiEventId = 2000;
+        private const int FetchingModelContentEventId = 2001;
+        private const int DiscoveredDependenciesEventId = 2002;
+        private const int SkippingPreprocessedDtmiEventId = 2003;
+        private const int InvalidDtmiInputEventId = 4000;
+        private const int ErrorFetchingModelContentEventId = 4004;
+        private const int IncorrectDtmiCasingEventId = 4006;
+
+        public static ResolverEventSource Shared { get; } = new ResolverEventSource();
+
+        private ResolverEventSource() : base(EventSourceName, EventSourceSettings.Default, AzureEventSourceListener.TraitName, AzureEventSourceListener.TraitValue) { }
+
+        [Event(InitFetcherEventId, Level = EventLevel.Informational, Message = StandardStrings.ClientInitWithFetcher)]
+        public void InitFetcher(Guid clientId, string scheme)
```

Comment:
We shouldn't do custom event writing, rather fit in with Azure.Core's diagnostic's strategy.

Reply:
Agreed. Task added